sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -1026,4 +1112,8 @@ case class FormatNumber(x: Expression, d: Expression)
   }
 
   override def prettyName: String = "format_number"
+
+  override def sql: Option[String] = for {
+    childrenSQL <- sequenceOption(children.map(_.sql))
+  } yield s"$prettyName(${childrenSQL.mkString(", ")})"
```

Comment:
Should we create a trait to implement sql string for all function like expressions? It's a lot of duplicated code here...